frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/common.ts

```diff
@@ -0,0 +1,51 @@
+import { TemplateKind } from '@console/internal/module/k8s';
+import { getName, getNamespace } from '@console/shared/src';
+import { VMSettingsField } from '../../../../components/create-vm-wizard/types';
+import { asSimpleSettings } from '../../../../components/create-vm-wizard/selectors/vm-settings';
+import {
+  ANNOTATION_DESCRIPTION,
+  LABEL_USED_TEMPLATE_NAME,
+  LABEL_USED_TEMPLATE_NAMESPACE,
+  TEMPLATE_FLAVOR_LABEL,
+  TEMPLATE_OS_LABEL,
+  TEMPLATE_OS_NAME_ANNOTATION,
+  TEMPLATE_WORKLOAD_LABEL,
+} from '../../../../constants/vm';
+import { MutableVMWrapper } from '../../../wrapper/vm/vm-wrapper';
+import { getTemplateOperatingSystems } from '../../../../selectors/vm-template/advanced';
+import { MutableVMTemplateWrapper } from '../../../wrapper/vm/vm-template-wrapper';
+import { operatingSystemsNative } from '../../../../components/create-vm-wizard/native/consts';
+import { CreateVMEnhancedParams } from './types';
+
+export const initializeCommonMetadata = (
+  { vmSettings, templates, openshiftFlag }: CreateVMEnhancedParams,
+  entity: MutableVMWrapper | MutableVMTemplateWrapper,
+  template?: TemplateKind,
+) => {
+  const settings = asSimpleSettings(vmSettings);
+  const operatingSystems = openshiftFlag
+    ? getTemplateOperatingSystems(templates)
+    : operatingSystemsNative;
+  const osID = settings[VMSettingsField.OPERATING_SYSTEM];
+  const osName = (operatingSystems.find(({ id }) => id === osID) || {}).name;
+
+  entity.addAnotation(`${TEMPLATE_OS_NAME_ANNOTATION}/${osID}`, osName);
+
+  if (settings[VMSettingsField.DESCRIPTION]) {
+    entity.addAnotation(ANNOTATION_DESCRIPTION, settings[VMSettingsField.DESCRIPTION]);
+  }
+
+  entity.addLabel(`${TEMPLATE_OS_LABEL}/${osID}`, 'true');
+  entity.addLabel(`${TEMPLATE_FLAVOR_LABEL}/${settings[VMSettingsField.FLAVOR]}`, 'true');
+  entity.addLabel(
+    `${TEMPLATE_WORKLOAD_LABEL}/${settings[VMSettingsField.WORKLOAD_PROFILE]}`,
+    'true',
+  );
+
+  if (template) {
+    entity.addLabel(LABEL_USED_TEMPLATE_NAME, getName(template));
+    entity.addLabel(LABEL_USED_TEMPLATE_NAMESPACE, getNamespace(template));
+  }
+
```

Comment:
we should add the `app: vm-name` label if it is missing one 
And probably `kubevirt.io/domain: vm-name` one as well, but into `spec.template.metadata.labels`

Reply:
ah, mistake: we should add this to the vm only - better to add these two behind  
```ts
vm.addTemplateLabel(TEMPLATE_VM_NAME_LABEL, vm.getName()); // for pairing service-vm (like for RDP)
```